### PR TITLE
RAB-1446: Fix Job and Step execution start time when a job resume

### PR DESCRIPTION
**Pull request by marinfazioakeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Tool/Component/Batch/Job/Job.php [hunk 1]
```diff
@@ -147,7 +149,9 @@ final public function execute(JobExecution $jobExecution): void
 
             $this->dispatchJobExecutionEvent(EventInterface::AFTER_JOB_EXECUTION, $jobExecution);
 
-            $jobExecution->setEndTime(new \DateTime());
+            if ($jobExecution->getStatus()->getValue() !== BatchStatus::PAUSED) {
```
Comment (StevenVAIDIE) [inline src/Akeneo/Tool/Component/Batch/Job/Job.php:152]:
```suggestion
            if (!$jobExecution->getStatus()->isPaused()) {
```

File: src/Akeneo/Tool/Component/Batch/Job/Job.php [hunk 2]
```diff
@@ -120,7 +120,9 @@ final public function execute(JobExecution $jobExecution): void
             $this->dispatchJobExecutionEvent(EventInterface::BEFORE_JOB_EXECUTION, $jobExecution);
```
Comment (StevenVAIDIE) [inline src/Akeneo/Tool/Component/Batch/Job/Job.php:121]: In order to do not imbricate `if` we could move it here, the only case we want to do is to set the start time when the job pass from starting to started, so we can do it here:
```
                if ($jobExecution->getStatus()->isStarting()) {
                    $jobExecution->setStartTime(new \DateTime());
                }
```

File: src/Akeneo/Tool/Component/Batch/Step/AbstractStep.php [hunk 3]
```diff
@@ -70,9 +70,10 @@ final public function execute(StepExecution $stepExecution)
 
         if ($stepExecution->getStatus()->getValue() === BatchStatus::PAUSED) {
             $this->dispatchStepExecutionEvent(EventInterface::BEFORE_STEP_EXECUTION_RESUME, $stepExecution);
+        } else {
+            $stepExecution->setStartTime(new \DateTime());
```
Comment (StevenVAIDIE) [inline src/Akeneo/Tool/Component/Batch/Step/AbstractStep.php:74]: We can move this line to src/Akeneo/Tool/Component/Batch/Job/Job.php at line 248 in order to just fill this value when we create a step execution not when we retrieve a job execution

**Review by StevenVAIDIE — APPROVED** [review]
If you want, you can rebase your 4 commit onto master, in this way you can merge this PR directly to master and do not wait on Export/Import resume PR. This PR does not really need it. By this way you don't need to create a EE, onboarder branch to made your CI pass

**Review by ValentinMumble — APPROVED** [review]

**Pull request merged**